### PR TITLE
[core] Replaced CGuard with corresponding lock: UniqueLock or ScopedLock

### PR DESCRIPTION
**Pull request by maxsharabayko**
Addresses an item in #1103.
`CGuard` was used as a temporal solution to avoid too many merge conflicts with the bonding functionality.
Now most of that functionality is merged, so it is relatively safe to replace `CGuard` with the corresponding lock: `ScopedLock` or `UniqueLock`.

### CODE REVIEW

File: srtcore/sync.h [hunk 1]
```diff
@@ -460,14 +454,14 @@ inline void releaseCond(Condition& cv) { cv.destroy(); }
 class CSync
 {
     Condition* m_cond;
-    CGuard* m_locker;
+    UniqueLock* m_locker;
 
 public:
-    // Locked version: must be declared only after the declaration of CGuard,
+    // Locked version: must be declared only after the declaration of ScopedLock,
```
Comment (ethouris) [inline srtcore/sync.h:460]: Here you should mentioned in the comment the same type that was used in the constructor parameters - `UniqueLock`

File: srtcore/sync.h [hunk 2]
```diff
@@ -534,7 +528,7 @@ class CSync
     // when you don't care whether the associated mutex is locked or not (you
     // accept the case that a mutex isn't locked and the signal gets effectively
     // missed), or you somehow know that the mutex is locked, but you don't have
-    // access to the associated CGuard object. This function, although it does
+    // access to the associated ScopedLock object. This function, although it does
```
Comment (ethouris) [inline srtcore/sync.h:531]: `UniqueLock`. You can't use `ScopedLock` in automated version of `CSync`.

You might improve this description by mentioning "`UniqueLock` or `Mutex` object".

File: srtcore/sync.h (hunk 1 above)
Comment (maxsharabayko) [reply srtcore/sync.h]: Fixed.

File: srtcore/sync.h (hunk 2 above)
Comment (maxsharabayko) [reply srtcore/sync.h]: Fixed.

**Pull request merged**